### PR TITLE
Adds Network Tables Source

### PR DESCRIPTION
**Pull request by JLLeitschuh**
This is currently a WIP.
At present it is impossible to load network tables on OSX as it wont link to the native library.
@ThomasJClark or @bradamiller Do you know why that it/what I may be doing wrong?

### CODE REVIEW

**Comment by ThomasJClark** [comment]
What error are you getting?  Is is only on OS X, or is that just the only platform you've tried?

**Comment by JLLeitschuh** [comment]
OSX is the only platform I've tried.

**Comment by ThomasJClark** [comment]
The CI builds got compile errors.  What error did you get?

**Comment by JLLeitschuh** [comment]
```java
platform: /Mac OS X/x86_64/

no ntcore in java.library.path
java.lang.UnsatisfiedLinkError: no ntcore in java.library.path
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1864)
	at java.lang.Runtime.loadLibrary0(Runtime.java:870)
	at java.lang.System.loadLibrary(System.java:1122)
	at edu.wpi.first.wpilibj.networktables.NetworkTablesJNI.<clinit>(NetworkTablesJNI.java:52)
	at edu.wpi.first.wpilibj.networktables.NetworkTable.initialize(NetworkTable.java:44)
	at edu.wpi.first.wpilibj.networktables.NetworkTable.getTable(NetworkTable.java:176)
	at edu.wpi.grip.core.sinks.NetworkTablesSinkTest.<init>(NetworkTablesSinkTest.java:17)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.junit.runners.BlockJUnit4ClassRunner.createTest(BlockJUnit4ClassRunner.java:217)
	at org.junit.runners.BlockJUnit4ClassRunner$1.runReflectiveCall(BlockJUnit4ClassRunner.java:266)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.BlockJUnit4ClassRunner.methodBlock(BlockJUnit4ClassRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:105)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:56)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:64)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:50)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:360)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by ThomasJClark** [comment]
Could you include the jars in your commit so I can test this?

**Comment by JLLeitschuh** [comment]
Yea, realized that was the problem for CI. Just pushed

**Comment by JLLeitschuh** [comment]
So this links correctly for travis: https://travis-ci.org/WPIRoboticsProjects/GRIP/builds/91326316
Windows doesn't actually run the tests at the moment (it should though) I'll open a PR for that soon.

File: src/main/java/edu/wpi/grip/core/sinks/NetworkTablesSink.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+public class NetworkTablesSink {
+    protected static final String TABLE_NAME = "GRIP";
+    /**
+     * The {@link NetworkTable} used by {@link NetworkTablesSink}
+     */
+    private static final NetworkTable table = NetworkTable.getTable(TABLE_NAME);
+
+    public NetworkTablesSink(EventBus eventBus) {
+        eventBus.register(this);
```
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/sinks/NetworkTablesSink.java]: How should I fix this for this PR?

Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/sinks/NetworkTablesSink.java]: What's the problem?

Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/sinks/NetworkTablesSink.java]: Nvm, you covered it in the gitter chat. Thanks.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#Tue Nov 17 16:35:37 EST 2015
+#Fri Nov 20 18:57:20 EST 2015
```
Comment (JLLeitschuh) [inline gradle/wrapper/gradle-wrapper.properties]: Remove this!

File: build.gradle [hunk 3]
```diff
@@ -35,7 +35,11 @@ repositories {
     jcenter()
 }
 
+def os = System.getProperty("os.name").toLowerCase()
 dependencies {
+    if (os.contains("windows")) compile fileTree(include: ['*windows.jar'], dir: 'libs')
+    else if (os.contains("linux")) compile fileTree(include: ['*linux.jar'], dir: 'libs')
+    else if (os.contains("mac")) compile fileTree(include: ['*os*x.jar'], dir: 'libs')
```
Comment (JLLeitschuh) [inline build.gradle]: Use the os dependency thingy.

File: build.gradle [hunk 4]
```diff
@@ -33,9 +33,14 @@ sourceSets {
 repositories {
     mavenCentral()
     jcenter()
+    maven {
+        url="http://first.wpi.edu/FRC/roborio/maven/"
```
Comment (JLLeitschuh) [inline build.gradle]: http://first.wpi.edu/FRC/roborio/maven/development

File: build.gradle [hunk 5]
```diff
@@ -33,9 +33,14 @@ sourceSets {
...
+    }
 }
 
 dependencies {
+    compile group: 'edu.wpi.first.wpilib.networktables.java', name: 'NetworkTables', version:'3.0.0-SNAPSHOT', classifier:'desktop-development'
```
Comment (JLLeitschuh) [inline build.gradle]: compile group: 'edu.wpi.first.wpilib.networktables.java', name: 'NetworkTables', version:'3.0.0-SNAPSHOT', classifier:'desktop'

**Pull request closed without merging**